Update libwebrtc RestoreEncodingLayers

youennf opened the pull request:
#### 0006fa5e6a5b10678251c393335e1d5d1072a305
<pre>
Update libwebrtc RestoreEncodingLayers
<a href="https://bugs.webkit.org/show_bug.cgi?id=242852">https://bugs.webkit.org/show_bug.cgi?id=242852</a>
rdar://97183357

Reviewed by Alex Christensen.

* Source/ThirdParty/libwebrtc/Source/webrtc/pc/rtp_sender.cc:

Canonical link: <a href="https://commits.webkit.org/252592@main">https://commits.webkit.org/252592@main</a>
</pre>

achristensen07 commented on Source/ThirdParty/libwebrtc/Source/webrtc/pc/rtp_sender.cc:
```diff
@@ -84,6 +84,7 @@ RtpParameters RestoreEncodingLayers(
       result.encodings.push_back(encoding);
       continue;
     }
+    RTC_CHECK_LT(index, parameters.encodings.size());
```
This seems redundant.  RTC_CHECK_EQ on line 77 does the same thing.

achristensen07 replied:
I guess there's no reason not to add this check, though.

youennf replied:
This is partially redundant but helps a bit.

youennf committed:
Update libwebrtc RestoreEncodingLayers
https://bugs.webkit.org/show_bug.cgi?id=242852
rdar://97183357

Reviewed by Alex Christensen.

* Source/ThirdParty/libwebrtc/Source/webrtc/pc/rtp_sender.cc:

Canonical link: https://commits.webkit.org/252592@main
